aop/src/main/java/io/micronaut/aop/InterceptedMethod.java

```diff
@@ -163,6 +163,27 @@ default Object unsupported() {
         throw new ConfigurationException("Cannot intercept method invocation, missing '" + resultType() + "' interceptor configured");
     }
 
+    /**
+     * Get native context from the suspension point.
+     *
+     * @return The native context from the suspension point or null
+     * @since 3.2
+     */
+    @Nullable
+    default Object getNativeContext() {
+        return null;
+    }
+
+    /**
+     * Update native context for the suspension point.
+     *
+     * @param context The native context
+     * @since 3.2
+     */
+    default void updateNativeContext(@Nullable Object context) {
+        throw new IllegalStateException("Updating native context not supported for interceptor: " + this);
+    }
```

Comment:
I think these should be placed behind an interface that you use an `instanceof` check on on the interceptor. Perhaps called it `SuspendedMethod`

Reply:
Added a new interface just for Kotlin methods